extensions/amp-analytics/0.1/events.js

```diff
@@ -1203,6 +1221,34 @@ export class VisibilityTracker extends EventTracker {
     });
   }
 
+  /**
+   * Returns a Promise indicating that we're ready to report the analytics,
+   * in the case of reportWhen: documentExit
+   * @return {!Promise}
+   * @private
+   */
+  createReportReadyPromiseForDocumentExit_() {
+    const deferred = new Deferred();
+    const root = this.root.getRoot();
+    let unloadListener, pageHideListener;
+    // Fired when document is being unloaded
```

Comment:
could you also explain the difference between unload and pagehide?

Reply:
I'm surprised by this, but according to the chart below, the only difference is that in Safari (mobile and desktop) when the user navigates to another page, pagehide will fire but unload will not. Elaborated the comment to reflect.